rmgpy/rmg/model.py

```diff
@@ -995,7 +1004,80 @@ def addSpeciesToEdge(self, spec):
         Add a species `spec` to the reaction model edge.
         """
         self.edge.species.append(spec)
-
+    
+    def setThermodynamicFilteringParameters(self,Tmax, toleranceThermoKeepSpeciesInEdge,minCoreSizeForPrune,maximumEdgeSpecies,reactionSystems):
+        """
+        sets parameters for thermodynamic filtering based on the current core
+        """
+        self.Tmax = Tmax
+        Gs = [spc.thermo.getFreeEnergy(Tmax) for spc in self.core.species]
+        self.Gmax = max(Gs)
+        self.Gmin = min(Gs)
+        
+        self.Gfmax = toleranceThermoKeepSpeciesInEdge*(self.Gmax-self.Gmin)+self.Gmax
+        self.toleranceThermoKeepSpeciesInEdge = toleranceThermoKeepSpeciesInEdge
+        self.minCoreSizeForPrune = minCoreSizeForPrune
+        self.reactionSystems = reactionSystems
+        self.maximumEdgeSpecies = maximumEdgeSpecies
+        
+    def thermoFilterDown(self,maximumEdgeSpecies,minSpeciesExistIterationsForPrune=0):
+        """
+        removes species from the edge based on their Gibbs energy until maximumEdgeSpecies
+        is reached under the constraint that all removed species are older than
+        minSpeciesExistIterationsForPrune iterations
```

Comment:
Sorry, I didn't say that in a way that made sense.  Unlike flux pruning we don't need a complete simulation to calculate our criteria, furthermore, running a simulation doesn't affect our criteria very much (since it's just calculated from the thermo of that edge species and the core thermo values).  Since thermodynamic filtering isn't strongly related to the simulation I don't see a compelling reason to make it dependent on how many simulations we've run.